core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -7378,6 +7368,37 @@ static class ExtendedExpander extends Expander {
       return super.visit(literal);
     }
 
+    /**
+     * Add all possible expandable 'group by' expression to set, which is
+     * used to check whether expr could be expanded as alias or ordinal.
+     */
+    @RequiresNonNull({"root"})
+    private void addExpandableExpressions(@UnknownInitialization ExtendedExpander this) {
+      switch (root.getKind()) {
+      case IDENTIFIER:
+      case LITERAL:
+        aliasOrdinalExpandSet.add(root);
+        break;
+      case GROUPING_SETS:
+      case ROLLUP:
+      case CUBE:
+        if (root instanceof SqlBasicCall) {
+          List<SqlNode> operandList = ((SqlBasicCall) root).getOperandList();
+          for (SqlNode node : operandList) {
+            if (node.getKind() == SqlKind.ROW) {
+              List<SqlNode> rowOperandList = ((SqlCall) node).getOperandList();
+              aliasOrdinalExpandSet.addAll(rowOperandList);
```

Comment:
Can't ROWS contain other ROWS as fields?
In that case do you need a recursive call?

Reply:
Thanks @mihaibudiu , I test on bigquery and spark, they behave as you mentioned, we should consider the recursive rows, I added a method to check the nested rows case, you can review it when you are available.